File: oving1/race.c
#include <pthread.h>
#include <stdio.h>

int i = 0;

void *thread1(){
	for(int x = 0; x < 1000000; x++){
		i++;
	}
	return 0;
}

void *thread2(){
	for(int x = 0; x < 1000000; x++){
		i--;
	}
	return 0;
}

int main(){
	pthread_t myThread1;
	pthread_t myThread2;

	pthread_create(&myThread1, NULL, thread1, NULL);
	pthread_create(&myThread2, NULL, thread2, NULL);	
	
	pthread_join(myThread1, NULL);

	

	pthread_join(myThread2, NULL);
	
	printf("%i\n", i);

	return 0;
}

File: oving1/race.py
from threading import Thread

i = 0;



def thread1():
	global i
	for x in range(1000000):
		i+= 1

def thread2():
	global i
	for x in range(1000000):
		i-= 1



def main():
	myThread1 = Thread(target = thread1, args = (),)
	myThread2 = Thread(target = thread2, args = (),)
	myThread1.start()
	myThread2.start()
	myThread1.join()
	myThread2.join()
	print(i)
	
main()